xpdan/vend/callbacks/broker.py

```diff
@@ -0,0 +1,320 @@
+import os
+import time as ttime
+from .core import CallbackBase, Retrieve
+from bluesky.utils import ensure_uid
+import numpy as np
+
+
+class BrokerCallbackBase(CallbackBase):
+    """
+    Base class for callbacks which need filled documents
```

Comment:
I think this needs a tad more explanation in a longer para below....for example, what is a "filled document"?  Why does the base class need this?  Is this broker base-class filling that need?  If so, it needs to say something like "extension to Callbackbase to solve such and such problem"